c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -1954,30 +1956,30 @@ c2_status_t MfxC2DecoderComponent::AllocateFrame(MfxC2FrameOut* frame_out, bool
                     std::shared_ptr<C2GraphicBlock> new_block;
                     bool found_surface = false;
 
-                    while (it != m_surfaces.end() && !found_surface) {
+                    while (!found_surface) {
                         // Buffer locked, try next block.
-                        res = AllocateC2Block(MFXGetSurfaceWidth(m_mfxVideoParams.mfx.FrameInfo, m_mfxVideoParams.IOPattern == MFX_IOPATTERN_OUT_VIDEO_MEMORY),
-                                      MFXGetSurfaceHeight(m_mfxVideoParams.mfx.FrameInfo, m_mfxVideoParams.IOPattern == MFX_IOPATTERN_OUT_VIDEO_MEMORY),
+                        res = AllocateC2Block(MFXGetSurfaceWidth(m_mfxVideoParams.mfx.FrameInfo, true),
+                                      MFXGetSurfaceHeight(m_mfxVideoParams.mfx.FrameInfo, true),
                                       m_mfxVideoParams.mfx.FrameInfo.FourCC, &new_block);
 
                         if (C2_TIMED_OUT == res) continue;
 
-                        if (C2_OK != res) break;
```

Comment:
both two breaks have been removed, then it seems the while loop is useless, right?

Reply:
> both two breaks have been removed, then it seems the while loop is useless, right?

When the correct surface allocated, found_surface is set true and while loop will end.

Reply:
ok